### PR TITLE
Add test to add and remove order notes.

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Adds an e2e test for adding and removing order notes.

Closes # https://github.com/woocommerce/woocommerce-quality/issues/603

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `pnpm env:start`
2. 'pnpm test:e2e-pw merchant/order-edit.spec.js`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js [hunk 1]
```diff
@@ -87,6 +87,40 @@ test.describe( 'Edit order', () => {
 		);
 	} );
 
+	test( 'can add and delete order notes', async ( { page } ) => {
+		// open order we created
+		await page.goto( `wp-admin/post.php?post=${ orderId }&action=edit` );
```
Comment (alopezari) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js:92]: Considering HPOS is enabled by default, should we start pointing these URLs to the new Orders URL?

**Review by rodelgc — APPROVED** [review]
Tested well as per the testing instructions 🎉  Thanks for working on this @lanej0 
Also tested the grep version: `pnpm playwright test  "-g" "can add and delete order notes" "--workers=1" "--repeat-each=2"`. Approving this PR but I'll leave it open in case anyone from the team would like to suggest any changes.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js (hunk 1 above)
Comment (nigeljamesstevenson) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js:92]: This is a great point. I have created https://github.com/woocommerce/woocommerce/issues/40846 for this which can be handled separately from this PR.

**Review by nigeljamesstevenson — APPROVED** [review]
Thanks @lanej0 - tested well!
I think we can handle the orders URL updates in a separate task : https://github.com/woocommerce/woocommerce/issues/40846

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js (hunk 1 above)
Comment (alopezari) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js:92]: Cool, thanks @nigeljamesstevenson! I will merge this 👍

**Pull request merged**